### ISSUE TITLE
chia-docker leads to network outages for other containers on the same server

### ISSUE THREAD

**Issue by m4tt075** [Chia-Network/chia-docker#144]
I've been experiencing issues with my TV-backend server since some weeks. Live-TV streams are regularly disrupted. The TV server runs in a docker container on my server.

I've now noticed that those disruptions are caused by the chia docker container running on the same server. When I stop the container, Live-TV works normally. When I start it, Live-TV works normally until the chia node is connected. Then, the disruptions start. 

The TV server and chia run on separate bridge networks and are not connected. 

Do you have any idea how I can solve the stiuation and run both containers at the same time?

**Comment by Starttoaster** [comment]
Based off of the information provided, the first thing that comes to mind is resource contention on a low power node. Try watching the output of `docker stats` to view resource usage of either container.

**Comment by Starttoaster** [comment]
@m4tt075 did you get time to look into this further? Also do you know what the Chia container is doing at the times that it's a problem (ie. is the full node in the process of getting up to sync?) I'm not familiar with Live-TV, is it a straight web stream or does it need to do some software transcoding of the stream for your clients?

**Comment by stefan-lange** [comment]
Sounds like the CPU is struggling, because of the spikes during the processing within in the full node. You can try to limit the CPUs for the container.

**Comment by Starttoaster** [comment]
> Sounds like the CPU is struggling, because of the spikes during the processing within in the full node. You can try to limit the CPUs for the container.

Indeed. This is pretty much what I was getting at. Was hoping to lead this user to that conclusion by following some simple troubleshooting steps to get their themselves 😌

**Comment by m4tt075** [comment]
Thanks for looking into this. You are most likely both right. I've restarted my Chia container. It's resyncing now (since two days ago). And using `docker stats` I can see that the CPU is maxing out at 100% and that this coincides with the streams stalling. I'll try to see how I can limit the CPUs for this container and whether it helps with the problem.

**Comment by m4tt075** [comment]
I'm still experiencing heavy buffering on my TV streams. As I stopped chia for some time, it keeps synchronizing, which apparently requires significant CPU load. On the other hand I'm not sure whether it is the CPU lead that is the limiting factor now.

Here a `docker stats`  snapshot while buffering occurs:

![image](https://user-images.githubusercontent.com/8303173/159349364-989e7d27-7ce2-4c6e-8289-eb3c4e7a0c10.png)

Block I/O seems "high", too, but I have no idea whether that is the cause now. Any recommendations how I should tweak my container priorities further from here?

**Comment by stefan-lange** [comment]
You could lower the connection count of the full node in the config. Do you have the default of `80`? I'm using 10, which works on RPi or a 10 years old i3-3225. Your machine seems to be more "beefy" :D. Try 40 or lower it further until you are happy. The connection handling in chia is what causing much cpu load, especially in case of many transactions aka 'dust storm'.

**Comment by m4tt075** [comment]
I've found `target_peer_count = 80` in my `config.yaml´. Is that the setting you are referring to?

**Comment by m4tt075** [comment]
I've tried it, but the issue stayed. I've scanned through the chia container log file again and found this:
"2022-03-21T22:51:34.974 full_node chia.full_node.coin_store: WARNING  Height 1649997: It took 13.48s to apply 1841 additions and 48 removals to the coin store. Make sure blockchain database is on a fast drive"

My setup: 
- Debian-based Openmediavault running on an M2 SSD
- HDD-based RAID5 hosting my files, including the "Appdata" folder for all my docker containers. This is where the chia db sits.
- Following the discussion you've linked, hosting the db on HDDs should be fine though

Any further advice appreciated!

**Comment by stefan-lange** [comment]
How long is your chia container up and running? It takes a hugh amount of memory. Do you have 1.2.11? If so, you could restart it. 1.2.11 seems to have some memory leaks.

This is mine: chia 1.3.1;
- the farmer (node + farmer, no wallet) has mem limit of 4GB and cpu limit of 2.
- the harvester has mem limit 256MB, cpu limit 2 (which is more than enough)
![grafik](https://user-images.githubusercontent.com/1436672/159375087-0c1909a0-9493-4407-adff-2ee272ab6510.png)

**Comment by m4tt075** [comment]
It hasn't been running for long, but was still synchronizing, which - I assume - consumed all the memory, when I took the snapshot. I'm running chia 1.3.1, like you do. 

I still have a M2 NVME left, but ran out of M2 slots on my server. I've ordered a PCI M2 adapter to reinstall it and will move Chia's `Appdata` folder there. If the HDDs are too slow, I hope this will finally fix my system again.

**Comment by stefan-lange** [comment]
Ah, ok, it was/is synching. Then the load is ok. You could limit CPUs to 1 and after initial sync back to the current value.

**Comment by Starttoaster** [comment]
> It hasn't been running for long, but was still synchronizing, which - I assume - consumed all the memory, when I took the snapshot. I'm running chia 1.3.1, like you do.
> 
> I still have a M2 NVME left, but ran out of M2 slots on my server. I've ordered a PCI M2 adapter to reinstall it and will move Chia's `Appdata` folder there. If the HDDs are too slow, I hope this will finally fix my system again.

If it doesn't completely fix things, I'd lean towards saying you may want to synchronize a Chia database on a stronger computer that you aren't using as a TV streaming server, and then once you have a fully synced database you might have better luck running it on your TV streaming server with lower system resource usage. But I'd also agree with @stefan-lange that storing the blockchain database on SSD is a huge quality of life improvement for a farm.

At this time, it seems like this is less of an issue with the chia-docker container and more a problem of resource budgeting in your current setup, so I'm just going to add a Help/Support tag to this Issue.

**Comment by m4tt075** [comment]
I hope you are right!  :-)

The PCI SSD adapter arrived. I've just moved the whole AppData folder, including the db, to the SSD. So far, so good, but I'll test and report back. If the issues are gone, I'll release some of the earlier resource restrictions again and see how it goes. But too early to celebrate. Will do some more testing...

**Comment by stefan-lange** [comment]
Nice 🙂

**Comment by Starttoaster** [comment]
Glad to hear it! Closing this Issue since it sounds like it's resolved.

**Comment by m4tt075** [comment]
To close the loop from my side: Moving the db to an SSD solved most of the issue. After taking away the resource limitations I had introduced before moving the db, I did encounter streaming issues again. Far less frequently than before, but still, when chia was synchronizing. 

As such, the solution is to combine a quick (SSD) hard drive AND resource restrictions for the docker container w.r.t. CPU and memory. 

Many thanks to @stefan-lange and @Starttoaster for leading the way.
